recipe/patches/0004-Support-numpy-2.0.patch

```diff
@@ -0,0 +1,38 @@
+diff --git a/libs/python/src/numpy/dtype.cpp b/libs/python/src/numpy/dtype.cpp
+index da30d192..1ce8c6ec 100644
+--- a/libs/python/src/numpy/dtype.cpp
++++ b/libs/python/src/numpy/dtype.cpp
+@@ -107,32 +107,7 @@ int dtype::get_itemsize() const {
+ }
+ 
+ bool equivalent(dtype const & a, dtype const & b) {
+-    // On Windows x64, the behaviour described on 
+-    // http://docs.scipy.org/doc/numpy/reference/c-api.array.html for
+-    // PyArray_EquivTypes unfortunately does not extend as expected:
+-    // "For example, on 32-bit platforms, NPY_LONG and NPY_INT are equivalent".
+-    // This should also hold for 64-bit platforms (and does on Linux), but not
+-    // on Windows. Implement an alternative:
+-#ifdef _MSC_VER
+-    if (sizeof(long) == sizeof(int) &&
+-        // Manually take care of the type equivalence.
+-        ((a == dtype::get_builtin<long>() || a == dtype::get_builtin<int>()) &&
+-         (b == dtype::get_builtin<long>() || b == dtype::get_builtin<int>()) ||
+-         (a == dtype::get_builtin<unsigned int>() || a == dtype::get_builtin<unsigned long>()) &&
+-         (b == dtype::get_builtin<unsigned int>() || b == dtype::get_builtin<unsigned long>()))) {
+-        return true;
+-    } else {
+-        return PyArray_EquivTypes(
+-            reinterpret_cast<PyArray_Descr*>(a.ptr()),
+-            reinterpret_cast<PyArray_Descr*>(b.ptr())
+-        );
+-    }
+-#else
+-    return PyArray_EquivTypes(
+-        reinterpret_cast<PyArray_Descr*>(a.ptr()),
+-        reinterpret_cast<PyArray_Descr*>(b.ptr())
+-    );
+-#endif
++  return a == b;
```

Comment:
Does this work for all numpy versions >=1.19? If not, we'll need to add a corresponding lower bound to the `run:` requirements of `libboost-python`.

Reply:
I have not tested other `numpy` versions, so setting the lower bound may be the safest option.

Reply:
Is there a way to set the minimum if `numpy` is an optional dependency? Alternatively, I can test my code for `numpy` versions less than 2.0 to check if the fix works generally.

Reply:
You can add a `run_constraints` constraint.